slice-core-api/src/main/java/com/cognifide/slice/api/tag/SliceTagUtils.java

```diff
@@ -74,6 +74,35 @@ private SliceTagUtils() {
 	}
 
 	/**
+	 * A helper method that returns the {@link Class} object, resolving it via it's Canonical Name.
+	 *
+	 * @param pageContext allows to access request context
+	 * @param type canonical name of the modal object, whose {@link Class} object needs to be returned
+	 * @return {@link Class} object pertaining to {@code type} as it's canonical name
+	 * @throws ClassNotFoundException if the class was not found
+	 */
+	public static Class<?> getClassFromType(final PageContext pageContext, final String type) throws ClassNotFoundException {
```

Comment:
Hmm.. By introducing of this method you're fetching the model twice during tag execution. Refactor it please, so that modelProvider#get(String, Resource) is used but model is read only once.